integration_test/ops_agent_test.go

```diff
@@ -1440,19 +1442,54 @@ func TestTCPLog(t *testing.T) {
 		if err := agents.SetupOpsAgent(ctx, logger.ToMainLog(), vm, config); err != nil {
 			t.Fatal(err)
 		}
+		command := `echo '{"msg":"test tcp log 1"}{"msg":"test tcp log 2"}' | /opt/google-cloud-ops-agent/subagents/fluent-bit/bin/fluent-bit -i stdin -o tcp://127.0.0.1:5170 -p format=json_lines && echo '{"msg":"test tcp log 3"}{"msg":"test tcp log 4"}' | /opt/google-cloud-ops-agent/subagents/fluent-bit/bin/fluent-bit -i stdin -o tcp://127.0.0.1:5170 -p format=json_lines`
+		// Installing Netcat or equivalent tools on Windows is tricky so we can instead
+		// use this Powershell Script that can send TCP messages.
+		if gce.IsWindows(platform) {
+			tcp_script := `param([string]$TargetIP = "", [int]$TargetPort = 0, [string]$Message = "")
+
+			try {
+				if ($TargetIP -eq "" ) { $TargetIP = read-host "Enter target IP address" }
+				if ($TargetPort -eq 0 ) { $TargetPort = read-host "Enter target port" }
+				if ($Message -eq "" ) { $Message = read-host "Enter message to send" }
+
+					$IP = [System.Net.Dns]::GetHostAddresses($TargetIP) 
+					$Address = [System.Net.IPAddress]::Parse($IP) 
+					$Socket = New-Object System.Net.Sockets.TCPClient($Address,$TargetPort) 
+					$Stream = $Socket.GetStream() 
+					$Writer = New-Object System.IO.StreamWriter($Stream)
+					$Message | % {
+						$Writer.WriteLine($_)
+						$Writer.Flush()
+					}
+					$Stream.Close()
+					$Socket.Close()
+
+				"??  Done."
+				exit 0 # success
+			} catch {
+				"?? Error in line $($_.InvocationInfo.ScriptLineNumber): $($Error[0])"
+				exit 1
+			}`
+
+			if err := gce.UploadContent(ctx, logger.ToMainLog(), vm, strings.NewReader(tcp_script), `C:\Program Files\Google\Cloud Operations\Ops Agent\send_tcp.ps1`); err != nil {
+				t.Fatalf("Error uploading TCP script: %v", err)
+			}
+
+			command = `(& "C:\Program Files\Google\Cloud Operations\Ops Agent\send_tcp.ps1" 127.0.0.1 5170 '{"msg":"test tcp log 1"}{"msg":"test tcp log 2"}') -and (& "C:\Program Files\Google\Cloud Operations\Ops Agent\send_tcp.ps1" 127.0.0.1 5170 '{"msg":"test tcp log 3"}{"msg":"test tcp log 4"}')`
+		}
 
 		// Write JSON test log to TCP socket via bash redirect, to get around installing and using netcat.
 		// https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Redirections
-		if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", "echo '{\"msg\":\"test tcp log 1\"}{\"msg\":\"test tcp log 2\"}' > /dev/tcp/localhost/5170"); err != nil {
-			t.Fatalf("Error writing dummy TCP log line: %v", err)
-		}
 
-		if err := gce.WaitForLog(ctx, logger.ToMainLog(), vm, "tcp_logs", time.Hour, "jsonPayload.msg:test tcp log 1"); err != nil {
-			t.Error(err)
+		if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", command); err != nil {
+			t.Fatalf("Error writing dummy TCP log line: %v", err)
 		}
 
-		if err := gce.WaitForLog(ctx, logger.ToMainLog(), vm, "tcp_logs", time.Hour, "jsonPayload.msg:test tcp log 2"); err != nil {
-			t.Error(err)
+		for i := 1; i <= 4; i++ {
```

Comment:
we should do these in parallel. Otherwise, it will wait ~7 minutes back to back, 4 times, stretching the test duration out to 28+ minutes needlessly. For an example of how to do this, see https://github.com/GoogleCloudPlatform/ops-agent/blob/69bacda8ce09e5d9a1d2e5e90f8f70fe8c539be0/integration_test/ops_agent_test.go#L1996

The pattern is:
```
var waitGroup sync.WaitGroup
for i := 1; i <= 4; i++ {
  i:= i // https://go.dev/doc/faq#closures_and_goroutines
 waitGroup.Add(1)
 go func () {
    defer waitGroup.Done()
    if err := gce.WaitForLog(...); err != nil {
      t.Error(err)
    }
  }()
} // end for loop
```

Reply:
Yep, fully agreed. Made the change to wait for the logs in parallel.